crates/holochain_state/src/validation_receipts.rs

```diff
@@ -76,11 +76,12 @@ impl ValidationReceipt {
         let signatures = if signatures.iter().any(Result::is_ok) {
             signatures.into_iter().filter_map(Result::ok).collect()
         } else {
-            return Err(signatures
-                .into_iter()
-                .filter_map(Result::err)
-                .next()
-                .expect("Must contain at least one error if there is no success"));
+            let r = match signatures.into_iter().filter_map(Result::err).next() {
+                Some(err) => Err(err),
+                // Weird that we got to here with no errors but seems to be happening in production.
```

Comment:
maybe log this as a warning? and it might be worth trying to reproduce this in a test

Reply:
Yeah this seems quite weird. I read the logic. We start with a non-empty list of validators, and map them into futures for signing, which returns Results. At no point do we filter out those Results. The only way to have an empty list of errors would be to have an empty list of signatures.

I agree at least some logging would be good. Maybe could log out some diagnostics so we can get a sense of what's happening. Like `validators.len()` and `signatures.len()` at least.

Reply:
So this is only happening in production, it doesn't show up in our tests and frankly this doesn't seem worthy of the time when we can just make the error impossible. What are we going to do with the logs?

Reply:
@freesig I think that this is just silently ignoring a case that we all think should *already* be impossible. So unless we can convince ourselves that the manifestation of this impossible case is not a problem, then I think we should treat it as a potential problem rather than interpreting it as `Ok`. But is there a reason why this isn't actually a problem?

Reply:
@maackle @steveeJ How's this?